.github/gha_monitor/gha_monitor/__main__.py

```diff
@@ -0,0 +1,109 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import logging
+import os
+
+from . import github
+from . import sns
+from datetime import datetime, timedelta
+from dateutil import parser, tz
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+# What time range to consider alerting on failures.
+TIME_WINDOW_BEGIN = datetime.now().astimezone(tz.UTC) - timedelta(hours=float(os.getenv('MONITOR_FREQ_IN_HOURS')))
+TIME_WINDOW_END = datetime.now().astimezone(tz.UTC)
+
+
+class GitHubActions(github.GitHubClient):
+    params = {
+        # Needed when using an API key
+        'github_username': os.getenv('github_username', None),
+        'github_password': os.getenv('github_password', None),
+        # Use from within an Action - ignored if username set
+        'token': os.getenv('GITHUB_TOKEN', None),
+        'repo_organization': os.getenv('GITHUB_REPO_ORG'),
+        'repo': os.getenv('GITHUB_REPO')
+    }
+
+
+class S2nNotices(sns.SNSClient):
+    params = {
+        'topic_arn': 'arn:aws:sns:us-west-2:024603541914:s2n_notices'
+    }
+
+
+def message_text():
+    """ Formatting for text message. """
+    return """
+s2n GitHub Action monitor notice
+State: {conclusion}
+Repo: {repo}
+GHA failure time: {time}
+Workflow name: {workflow_name}
+URL: {url}
+started by: {commit_owner}\n
+"""
```

Comment:
Design thought: how do you intend to consume these? Mainly: are these meant to be consumed by a human or a machine? If the later, it would be better to use a more machine-friendly format (say: JSON?)

Reply:
Agreed- would like todo both a json and text format- looking at incorporating this and the other suggestions into another iteration.